### DISCUSSION TITLE
OpenBLAS MKL Library implementation

### DISCUSSION THREAD

**Discussion by MGN-GIT** [OpenMathLib/OpenBLAS#4997] [Q&A]
Hello Champs

How OpenBLAS GEMM3M implementation is different from intel's MKL implementation? What are the supported machine configurations for this implementation? What happends if these implementation are not supported on certain machine? Is there any alternative?
Thanks

**Comment by martin-frbg** [comment]
MKL is closed source, so how would anyone outside Intel know and tell ? If anything you can look at and compare to BLIS or publications on the "3M" algorithm. OpenBLAS currently has assembly implementations for most x86/x86_64 machines as well as for mips64/loongarch64 - outside these architectures the symbols would simply be missing from the library. Only when the assembly implementation cannot be compiled in (this practically only concerns compilation with the MSVC compiler) on a "supported" architecture you will currently run into serious problems, as the fallback for this case was never correctly implemented. This should be fixed soon.
In general, the alternative for GEMM3M would be the corresponding GEMM operation, as the two are functionally equivalent - though GEMM3M is "usually" faster. May I ask what is the background for your question ?
